src/method.jl

```diff
@@ -92,17 +92,16 @@ function signature(orig_sig::Type{<:Tuple}; extra_hygiene=false)
     return def
 end
 
-
-
 function slot_names(m::Method)
     ci = Base.uncompressed_ast(m)
     return ci.slotnames
 end
 
 function argument_names(m::Method)
     slot_syms = slot_names(m)
-    @assert slot_syms[1] === Symbol("#self#")
-    arg_names = slot_syms[2:m.nargs]  # nargs includes 1 for `#self#`
+    # nargs includes 1 for `#self#` or the function object name;
+    # TODO: could support option to return function object argument name/type
```

Comment:
I don't think we will want to do that, because this function is called `argument_names`
So getting the name of the function object itself is outside it's purpose
```suggestion
```